### ISSUE TITLE
Conduct an impact analysis of Inbound SMTP DANE with DNSSEC for Exchange Online

### ISSUE THREAD

**Issue by tkol2022** [cisagov/ScubaGear#1394]
# 💡 Summary #

Microsoft announced support in Exchange Online for two security standards: DNS-based Authentication of Named Entities (DANE) for SMTP and Domain Name System Security Extensions (DNSSEC). The scope of this issue is to:

- [x] Understand the new EXO features as described in the documentation

- [x] Perform a hands-on experiment with the new features

- [x] Should Scuba create a new policy or update an existing policy that includes the new features?

- [x] Can a policy check be codified in ScubaGear?

**Comment by adhilto** [comment]
For reference: https://techcommunity.microsoft.com/t5/exchange-team-blog/announcing-general-availability-of-inbound-smtp-dane-with-dnssec/ba-p/4281292

**Comment by adhilto** [comment]
Initial analysis:

TLDR:
- Is there value in enabling this feature in Exchange Online? **Yes**
- Could ScubaGear check for this feature? **Yes**
- How hard would it be for users to enable this feature? **Potentially quite hard. Historically speaking DNSSEC and DANE are hard to deploy correctly.**

The long version:

**Background:**
- DNSSEC: An extension to DNS that adds cryptographic integrity assurances to protect against forged responses (the original DNS protocol was/is actually quite vulnerable). This is actually a fairly old standard but unfortunately has had very slow adoption (similar to IPv6). For example, in 2017, a [study](https://www.usenix.org/conference/usenixsecurity17/technical-sessions/presentation/chung) showed that only around 1% of popular domains supported DNSSEC and of those about a third were misconfigured.
- SMTP: The basic email protocol. Of particular relevance for this issue: "out-of-the-box" it sends emails over plaintext. SMTP traffic can be encrypted using TLS, but only if both the sender and the receiver support it and are able to successfully communicate that they support it. And as this negotiation happens over cleartext, it's vulnerable to [downgrade attacks](https://datatracker.ietf.org/doc/html/rfc7672#section-1.3.1).
- DANE: A potential solution to the TLS downgrade problem. The basic idea: domains can publish their TLS certificates as DNS records, so senders can know with high-confidence that 1) the receiver supports TLS and 2) what TLS certificate to expect.

**DANE as it relates to Exchange Online:**
- If you're sending an email and the receiver supports DANE, Exchange will use DANE to secure the email. There is no action you need to take to enable this.
- The ability to secure inbound emails using DANE is the new feature discussed in the article.

**To enable inbound DANE, you must:**
1. Sign the zone of each accepted domain using DNSSEC. This step is done outside of the admin center. This is a step many users will find challenging. However, it's worth noting that per [OMB M-08-23](https://www.whitehouse.gov/wp-content/uploads/legacy_drupal_files/omb/memoranda/2008/m08-23.pdf), DNSSEC is already mandated for federal agencies.
2. Run the following PowerShell command for each accepted domain: `Enable-DnssecForVerifiedDomain -DomainName <DomainName>`
3. Change the `MX` record for that domain to the value returned by that command. Note that are a few additional steps needed to test the change rather than simply abruptly changing it, described [here](https://learn.microsoft.com/en-us/purview/how-smtp-dane-works#set-up-inbound-smtp-dane-with-dnssec).
4. Run the following PowerShell command for each accepted domain: `Enable-DnssecForVerifiedDomain -DomainName <DomainName>`

**Challenges with and limitations of this feature:**
- Assumes that you can deploy DNSSEC for all of your domains, something that historically has been quite challenging.
- DANE cannot yet be enabled for the *.onmicrosoft.com domain.
- Third-party gateways: if you use a third-party / on-prem gateway that receives email then forwards to M365, enabling DANE will be more complicated (but still possible). Same goes for third-party outbound gateways.
-  Fully delegated domains (hosted by Microsoft) do not yet support DANE. Neither are viral/self-service domains.
- Added complexity in ensuring the certificate used by the SMTP server and the one DNS record stay in sync (see [Under the Hood of DANE Mismanagement in SMTP](https://www.usenix.org/system/files/sec22summer_lee.pdf)). Though in this case, unless you're using a third-party gateway, Microsoft should handle any key roll-overs for you.

**Could ScubaGear check for DANE?**
- Yes!
- See https://learn.microsoft.com/en-us/powershell/module/exchange/get-dnssecstatusforverifieddomain?view=exchange-ps.

**Comment by adhilto** [comment]
> [@adhilto](https://github.com/adhilto) Is there any additional work to be done here or presented to others? Any follow on tasks? If not, I'd recommend closing as completed. If so, I'd recommend adding new issues to support the follow-on efforts.

No additional work to present. All that's missing from the acceptance criteria is a decision on the question, "Should Scuba create a new policy or update an existing policy that includes the new features?"
